KOGITO-7154 DMN algo for OAS/Swagger naming (#4224)
kiegroup/kie-docs: doc-content/kogito-docs/src/main/asciidoc/release-notes/ReleaseNotesKogito.1.24.0.Final/ReleaseNotesKogito.1.24.0.Final.adoc

// IMPORTANT: For 1.10 and later, save each version release notes as its own module file in the release-notes folder that this `ReleaseNotesKogito<version>.adoc` file is in, and then include each version release notes file in the chap-kogito-release-notes.adoc after Additional resources of {PRODUCT} deployment on {OPENSHIFT} section, in the following format:
//include::ReleaseNotesKogito.<version>/ReleaseNotesKogito.<version>.adoc[leveloffset=+1]

[id="ref-kogito-rn-new-features-1.24_{context}"]
= New features in {PRODUCT} 1.24

[role="_abstract"]
The following sections describe some of the new features or enhancements in {PRODUCT} 1.24.

== {PRODUCT} runtimes

=== {PRODUCT} change in DMN/Decisions code generated OAS/Swagger type names

In previous versions of {PRODUCT}, model-specific OAS/Swagger type names generated from DMN/Decisions models used a number-based format, for example `InputSet1`, `InputSet2`, etc.

> However, this format has some limitations impacting on external clients and external plugins, across evolutions of the Kogito application being developed; for example, adding/removing DMN/Decisions models from your Kogito application does not guarantee a stable naming and ordering across evolutions of your Kogito application.

Reviewer: ```suggestion
However, this format has some limitations impacting external clients and external plugins across evolutions of the Kogito application being developed; For example, adding or removing DMN/Decisions models from your Kogito application does not guarantee a stable naming and ordering across evolutions of your Kogito application.
```

Author: ```suggestion
However, this format has some limitations impacting external clients and external plugins across evolutions of the Kogito application being developed. For example, adding or removing DMN/Decisions models from your Kogito application does not guarantee a stable naming and ordering across evolutions of your Kogito application.
```

per above, but I assume you wanted to replace `;` with `.` too, given the capitalisation of the "For example"

> Starting from {PRODUCT} 1.24.0, the model-specific OAS/Swagger type names by default will now be based on the names of the DMN/Decisions models themselves, for example `InputSetLoan`, `InputSetValidation`, etc.

Reviewer: ```suggestion
Starting from {PRODUCT} 1.24.0, the model-specific OAS/Swagger type names by default will now be based on the names of the DMN/Decisions models themselves, for example, `InputSetLoan`, `InputSetValidation`, etc.
```

> In the case two DMN/Decisions models share the same name, it will automatically fallback to the previous number-based format; in addition, the number-based format strategy now accounts for the DMN/Decisions models to be ordered by their namespace and name, to maximise stable naming results in the OAS/Swagger definitions file.

Reviewer: ```suggestion
In the case of two DMN/Decisions models sharing the same name, it will automatically fall back to the previous number-based format. In addition, the number-based format strategy now accounts for the DMN/Decisions models to be ordered by their namespace and name, to maximize stable naming results in the OAS/Swagger definitions file.
```

However, this format has some limitations impacting external clients and external plugins across evolutions of the Kogito application being developed. For example, adding or removing DMN/Decisions models from your Kogito application does not guarantee a stable naming and ordering across evolutions of your Kogito application.

Starting from {PRODUCT} 1.24.0, the model-specific OAS/Swagger type names by default will now be based on the names of the DMN/Decisions models themselves, for example, `InputSetLoan`, `InputSetValidation`, etc.
In the case of two DMN/Decisions models sharing the same name, it will automatically fall back to the previous number-based format. In addition, the number-based format strategy now accounts for the DMN/Decisions models to be ordered by their namespace and name, to maximize stable naming results in the OAS/Swagger definitions file.

NOTE
====
No change is required from a Kogito application development perspective.
====

This is an effective convenient improvement while integrating your Kogito-based application with external tools, clients, and plugins. For more information, see https://issues.redhat.com/browse/KOGITO-7154[KOGITO-7154].

> For more information, see https://issues.redhat.com/browse/KOGITO-7154[KOGITO-7154].

Author: ```suggestion
```

ended up being repeated, will remove.
